### DISCUSSION TITLE
onlyRenderVisibleElements. Do not apply for nodes of some type/id

### DISCUSSION THREAD

**Discussion by egorov-evrone** [xyflow/xyflow#4599] [Q&A]
Greetings! 

My application have graphs with custom nodes that make API requests (some of them). Those requests are made by Apollo useQuery hook. 

It was working properly until I enabled onlyRenderVisibleElements parameter. 

With this parameter enabled, when one opens a graph and scrolls away or zooms in - nodes, expectedly, unmounted. Therefore the API requests that were in the process get cancelled. 

My application needs those requests to finish and subsequent code to get executed.

The question is: is there a way to mark some of the nodes as "always visible"? So they won't get hidden/unmounted.

**Comment by bcakmakoglu** [comment]
>  is there a way to mark some of the nodes as "always visible"? So they won't get hidden/unmounted.

That's not possible using `onlyRenderVisibleElements` since it doesn't allow specifying what nodes should or shouldn't be unmounted whenever they leave the visible viewport.

You'd either have to implement this sort of logic yourself or just avoid using `onlyRenderVisibleElements`.
